nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureDataLakeStorageProcessor.java

```diff
@@ -158,41 +173,51 @@
     public static Collection<ValidationResult> validateCredentialProperties(final ValidationContext validationContext) {
         final List<ValidationResult> results = new ArrayList<>();
 
-        final boolean useManagedIdentity = validationContext.getProperty(USE_MANAGED_IDENTITY).asBoolean();
-        final boolean accountKeyIsSet  = validationContext.getProperty(ACCOUNT_KEY).isSet();
-        final boolean sasTokenIsSet     = validationContext.getProperty(SAS_TOKEN).isSet();
-
-        int credential_config_found = 0;
-        if(useManagedIdentity) credential_config_found++;
-        if(accountKeyIsSet) credential_config_found++;
-        if(sasTokenIsSet) credential_config_found++;
-
-        if(credential_config_found == 0){
-            final String msg = String.format(
-                "At least one of ['%s', '%s', '%s'] should be set",
-                ACCOUNT_KEY.getDisplayName(),
-                SAS_TOKEN.getDisplayName(),
-                USE_MANAGED_IDENTITY.getDisplayName()
-            );
-            results.add(new ValidationResult.Builder().subject("Credentials config").valid(false).explanation(msg).build());
-        } else if(credential_config_found > 1) {
-            final String msg = String.format(
-                "Only one of ['%s', '%s', '%s'] should be set",
-                ACCOUNT_KEY.getDisplayName(),
-                SAS_TOKEN.getDisplayName(),
-                USE_MANAGED_IDENTITY.getDisplayName()
-            );
-            results.add(new ValidationResult.Builder().subject("Credentials config").valid(false).explanation(msg).build());
+        if (!validationContext.getProperty(AzureStorageUtils.STORAGE_CREDENTIALS_SERVICE).isSet()) {
```

Comment:
When the CS defined, the processor level credential properties should not be defined. It should be validated.
Another option would be to get rid of the processor level credential properties. That would make the validation easier and the whole code simpler.

Reply:
I think it would make sense to have the credentials configurable only via the controller service.
I see the newer processors follow this pattern as well.

@MuazmaZ, would you be okay with removing all the Azure credentials-related properties from the processors and move them to the controller service?

Reply:
@tpalfy makes sense to me especially if that is the pattern for the newer processor.

Reply:
Great, thanks, I did the change.